trio_websocket/_impl.py

```diff
@@ -684,10 +748,17 @@ class WebSocketConnection(trio.abc.AsyncResource):
 
     CONNECTION_ID = itertools.count()
 
-    def __init__(self, stream, ws_connection, *, host=None, path=None,
-        client_subprotocols=None, client_extra_headers=None,
-        message_queue_size=MESSAGE_QUEUE_SIZE,
-        max_message_size=MAX_MESSAGE_SIZE):
+    def __init__(
+            self,
+            stream: trio.SocketStream | trio.SSLStream[trio.SocketStream],
```

Comment:
using pipe for union is python 3.10

doesn't mypy complain about these if we run the type check under 3.8?

Reply:
It works because I added `from __future__ import annotations`